internal/backend/user.go

```diff
@@ -52,9 +54,14 @@ func newUser(ctx context.Context, userID string, db *DB, remote *remote.User, st
 		defer user.updateWG.Done()
 		labels := pprof.Labels("go", "Connector Updates", "UserID", user.userID)
 		pprof.Do(ctx, labels, func(_ context.Context) {
-			for update := range remote.GetUpdates() {
-				if err := user.apply(context.Background(), update); err != nil {
-					logrus.WithError(err).Errorf("Failed to apply update: %v", err)
+			for {
+				select {
+				case update := <-remote.GetUpdates():
```

Comment:
Can we call `GetUpdates` outside the `for` loop, hold the channel in a variable, and then use it inside the for loop? I don't like that we repeatedly call `GetUpdates` each iteration; the implementation of the connector might construct a new channel each call, for instance.

Reply:
That would be a terrible connector, but I see your point. Fixed.